### PR TITLE
feat(api,sdk): Fix some issue with error wrap

### PR DESCRIPTION
**Pull request by richardlt**
- Add cause in http error message
- Replace all compare to sdk.Err by sdk.ErrorIs
- Create empty error and fmt.Errorf when wrapping sdk.Err to prevent
message duplication
- Improve error log for some import handlers

1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/api/workflow/dao_node_run_vulnerability.go [hunk 1]
```diff
@@ -104,15 +104,15 @@ func createNewVulnerabilityReport(db gorp.SqlExecutor, cache cache.Store, proj *
 
 	// Get summary from previous run
 	previousRunReport, err := loadPreviousRunVulnerabilityReport(db, nr)
-	if err != nil && err != sdk.ErrNotFound {
+	if err != nil && sdk.ErrorIs(err, sdk.ErrNotFound) {
```
Comment (richardlt) [inline engine/api/workflow/dao_node_run_vulnerability.go]: fix

File: engine/api/workflow/dao_node_run_vulnerability.go [hunk 2]
```diff
@@ -104,15 +104,15 @@ func createNewVulnerabilityReport(db gorp.SqlExecutor, cache cache.Store, proj *
...
 		return sdk.WrapError(err, "Unable to get previous vulnerability report")
 	}
 	nodeRunReport.Report.PreviousRunSummary = previousRunReport
 
 	// Get summary from default branch
 	if defaultBranch != "" && defaultBranch != nr.VCSBranch {
 		defaultBranchReport, err := loadLatestRunVulnerabilityReport(db, nr, defaultBranch)
-		if err != nil && err != sdk.ErrNotFound {
+		if err != nil && sdk.ErrorIs(err, sdk.ErrNotFound) {
```
Comment (richardlt) [inline engine/api/workflow/dao_node_run_vulnerability.go]: fix

File: sdk/exportentities/common.go [hunk 3]
```diff
@@ -9,6 +9,7 @@ import (
 	"strings"
 	"time"
 
+	"github.com/ovh/cds/sdk"
```
Comment (sguiheux) [inline sdk/exportentities/common.go]: wrong place

File: engine/api/environment_import.go [hunk 4]
```diff
@@ -71,13 +69,14 @@ func (api *API) postEnvironmentImportHandler() service.Handler {
...
 		_, msgList, globalError := environment.ParseAndImport(tx, api.Cache, proj, eenv, force, project.DecryptWithBuiltinKey, getUser(ctx))
 		msgListString := translate(r, msgList)
-
 		if globalError != nil {
-			myError, ok := globalError.(sdk.Error)
-			if ok {
-				return service.WriteJSON(w, msgListString, myError.Status)
+			globalError = sdk.WrapError(globalError, "Unable to import environment %s", eenv.Name)
+			if sdk.ErrorIs(globalError, sdk.ErrUnknownError) {
+				return globalError
```
Comment (fsamin) [inline engine/api/environment_import.go]: it's not exactly the same thing here...

File: engine/api/pipeline_import.go [hunk 5]
```diff
@@ -120,27 +119,27 @@ func (api *API) importPipelineHandler() service.Handler {
...
 
 		_, allMsg, globalError := pipeline.ParseAndImport(tx, api.Cache, proj, payload, getUser(ctx), pipeline.ImportOptions{Force: forceUpdate})
 		msgListString := translate(r, allMsg)
-
 		if globalError != nil {
-			myError, ok := globalError.(sdk.Error)
-			if ok {
-				return service.WriteJSON(w, msgListString, myError.Status)
+			globalError = sdk.WrapError(globalError, "Unable to import pipeline")
+			if sdk.ErrorIs(globalError, sdk.ErrUnknownError) {
```
Comment (fsamin) [inline engine/api/pipeline_import.go]: we have to discuss about it

File: engine/api/environment_import.go (hunk 4 above)
Comment (richardlt) [reply engine/api/environment_import.go]: I created a sdk.ErrorIsUnknown and updated the ErrorIs to return true is an error not known (not from our sdk) is compared with ErrUnknownError.

**Comment by ovh-cds** [comment]
CDS Report ut-engine#6711.0 ✘  
* Stage 1
  * Engine Test with DB ✘

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#6711.0 ✘  
* Build minimal
  * cdsctl linux ✘  
  * Engine Build linux amd64 ✘  
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✘

**Pull request merged**